core/embed/util/image/inc/util/boot_header.h

```diff
@@ -0,0 +1,203 @@
+/** Number of PQ/EC signatures */
+#define BOOT_HEADER_SIGNATURE_COUNT 2
+/** Length of PQ signature in bytes */
+#define BOOT_HEADER_PQ_SIGNATURE_LEN (7856)
+/** Length of EC signature in bytes */
+#define BOOT_HEADER_EC_SIGNATURE_LEN (64)
+/** Number of reserved slots for Merkle proof */
+#define BOOT_HEADER_MERKLE_PROOF_MAXLEN (256)
+/** Maximum accepted size of the bootloader code in bytes */
+#define BOOT_HEADER_CODE_MAXSIZE (512 * 1024)
+
+/**
+ * 4-byte version structure used in the boot header
+ */
+typedef struct __attribute__((packed)) {
+  uint8_t major;
+  uint8_t minor;
+  uint8_t patch;
+  uint8_t build;
+} boot_header_version_t;
+
+/**
+ * Merkle proof node (SHA-256 digest)
+ */
+typedef struct {
+  uint8_t bytes[32];
+} merkle_proof_node_t;
+
+/**
+ * Authenticated part of the boot header
+ *
+ * This structure can be extended in future versions if needed.
+ * Just make sure to add new fields at the end of the structure.
+ * Never remove or reorder existing fields.
+ */
+typedef struct __attribute__((packed)) {
+  /** Magic constant 'TRZQ' */
+  uint32_t magic;
+  /** Hardware model, e.g. 'T3W1'. */
+  uint32_t hw_model;
+  /** Hardware revision, e.g. 1 */
+  uint32_t hw_revision;
+  /** Bootloader version */
+  boot_header_version_t version;
+  /** Minimum version that the device can be downgraded to without
+   * erasing storage. */
+  boot_header_version_t fix_version;
+  /** Minimum previous version that the device can be updated from when
+   * installing this header. */
+  boot_header_version_t min_prev_version;
+  /** An integer which must not decrease between updates. */
+  uint8_t monotonic_version;
+  /** Bitmask of keys used for signature verification.
+   * Each bit corresponds to a public key in the BOOTLOADER_PQ_KEY and
+   * BOOTLOADER_EC_KEY arrays. If the bit is set, the corresponding key
+   * is used for signature verification. */
+  uint8_t sigmask;
+  /* Reserved bytes (alignment) */
+  uint8_t reserved[2];
+  /** Size of the entire header in bytes, including the Merkle proof
+   * and signatures. It's calculated at link time and must be
+   * aligned to 8K boundary. */
+  uint32_t header_size;
+  /** Size of the authenticated part of the header in bytes.
+   * Final value is calculated in post-build step and includes
+   * potential padding of the structure. */
+  uint32_t auth_size;
+  /** Size of the bootloader code in bytes */
+  uint32_t code_size;
+  /** Address of storage area for storage relocation purposes */
+  uint32_t storage_address;
+  /* Firmware Merkle tree root */
+  merkle_proof_node_t firmware_root;
+  /* Padding is automatically added by the post-build step to ensure that
+   * the authenticated part of the header is maximized. */
+  uint8_t padding[0];
+
+} boot_header_auth_t;
+
+/**
+ * Merkle proof structure used in the boot header to calculate the root
+ * of the Merkle tree. It is placed just after the authenticated part
+ * of the boot header.
+ */
```

Comment:
I suggest we add a warning here:
```suggestion
 * WARNING: This struct must not change, otherwise old boardloaders will
 * not interpret the header correctly. Any new unauthenticated fields
 * must be added at the end of `boot_header_unauth_t`.
 */
```

Reply:
added, see bc1f587246851844dd000174454c7810aab22725